src/main/flight/imu.c

```diff
@@ -428,7 +428,7 @@ static void imuCalculateEstimatedAttitude(timeUs_t currentTimeUs)
     }
 #endif
 #if defined(USE_GPS)
-    else if (STATE(FIXED_WING) && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat >= 5 && gpsSol.groundSpeed >= 300) {
+    if (!useMag && STATE(FIXED_WING) && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat >= 5 && gpsSol.groundSpeed >= 300) {
```

Comment:
Another possibility is:
```
if(0) { }
#ifdef USE_MAG
else if (sensors(SENSOR_MAG) && compassIsHealthy()) {

Reply:
Fortunately we had `useMag` with the correct logic in this case so I think it makes it much cleaner.  If not then your suggestion would be okay, but still seems a little cryptic.  The only other option would likely be repeating code in separate `#define` sections.